src/openrct2/management/NewsItem.h

```diff
@@ -52,14 +52,63 @@ struct NewsItem
     uint16_t MonthYear;
     uint8_t Day;
     utf8 Text[256];
+
+    constexpr bool IsEmpty() const noexcept
+    {
+        return Type == NEWS_ITEM_NULL;
+    }
 };
 
 constexpr int32_t NEWS_ITEM_HISTORY_START = 11;
-constexpr int32_t MAX_NEWS_ITEMS = 61;
+constexpr int32_t MAX_OLD_NEWS_ITEMS = 50;
+constexpr int32_t MAX_NEWS_ITEMS = NEWS_ITEM_HISTORY_START + MAX_OLD_NEWS_ITEMS;
 
 extern const uint8_t news_type_properties[10];
 
-extern NewsItem gNewsItems[MAX_NEWS_ITEMS];
+struct NewsItemQueue
+{
+    NewsItem& operator[](size_t index);
+    NewsItem* At(int32_t index);
+    bool IsEmpty() const;
+    void Init();
+    uint16_t IncrementTicks();
+    NewsItem& Current();
+    const NewsItem& Current() const;
+    NewsItem& Oldest();
+    const NewsItem& Oldest() const;
+    bool IsCurrentOld() const;
+    void MoveCurrentToOld();
+    NewsItem* FirstOpenOrCreateSlot();
+
+    template<typename Predicate> void ForeachRecentNews(Predicate&& p)
+    {
+        for (auto& newsItem : Recent)
+        {
+            if (newsItem.IsEmpty())
+                break;
+            p(newsItem);
+        }
+    }
+
+    template<typename Predicate> void ForeachOldNews(Predicate&& p)
+    {
+        for (auto& newsItem : Old)
+        {
+            if (newsItem.IsEmpty())
+                break;
+            p(newsItem);
+        }
+    }
+
+private:
+    int32_t RemoveTime() const;
+    void AppendToOld(NewsItem& item);
+
+    NewsItem Recent[NEWS_ITEM_HISTORY_START];
+    NewsItem Old[MAX_OLD_NEWS_ITEMS];
```

Comment:
I think these should be changed to vectors. All uses of `NEWS_ITEM_HISTORY_START` and `MAX_OLD_NEWS_ITEMS` should be removed with the exception of internal limits to ensure it never adds more than the current limit,  `NEWS_ITEM_HISTORY_START` and `MAX_OLD_NEWS_ITEMS` to the vectors.

Reply:
The objective of this change is exactly to allow us to change these away from C-arrays (as I mentioned in https://github.com/OpenRCT2/OpenRCT2/pull/11715#issue-415850000).

That is the next thing I'll do once this PR is approved :)